### ISSUE TITLE
Sort/deduplicate dependencies in flatten product

### ISSUE THREAD

**Issue by Anton-Latukha** [shellfire-dev/fatten#2]
For example flattened `overdrive` produces numbers of unsorted and duplicated occurrences **in**:

```bash
_core_dependency_requires
_core_dependency_internalDataSource_Alpine
_core_dependency_internalDataSource_CentOS
_core_dependency_internalDataSource_Debian
...
```

This is convenience. To see full list of required packages or binaries for distribution.
And also can shave-off couple hundreds of code from products.

**Comment by raphaelcohn** [comment]
Yes, it would be good idea, although the actual cost of this code (in bytes) is quite small. Do you have a suggestion for how?

More recently, I've ceased to use `flatten`, as I find it a little brittle, and just prefer to ship git repos as is... That, of course, wouldn't be to everyone's taste.

**Comment by Anton-Latukha** [comment]
No. I just pointed-out obvious thing that fallen in my view. Ans at first sight are very easy to solve, at first sight.

Than I read into code, how that array are used, and it revolves arround:

```bash
core_dependency_requires () 
{ 
    _local_packageManager="$1";
    if [ -z "$_local_packageManager" ]; then
        core_exitError 70 "packageManager can not be empty";
    fi;
    shift 1;
    for _local_programName in "$@";
    do
        if [ -z "$_local_programName" ]; then
            core_exitError 70 "programName can not be empty";
        fi;
        _core_dependency_requires="${_core_dependency_requires}${_local_packageManager} ${_local_programName}
";
        core_message DEBUG "Requires in packageManager '$_local_packageManager' program '$_local_programName'";
    done;
    unset _local_packageManager;
    unset _local_programName
}
```
And trying to understand `core_dependency_requires ()`, I see that this function is not used anywhere in the code. That is where my deduction stuck...

More, in the next issue 8)
To not to spread topic here.

**Comment by Anton-Latukha** [comment]
I am not thought about solutions, besides that `uniq` and `sort` are already in GNU Core Utilities, and they solve 95% of it.

Then `sed` or `awk` can be used by `flatten` to reattach first line of values to variable, but they not because of that required by resulting application.

It there is no way to not attach first line, maybe this:
```bash
_core_dependency_requires="\
* base64
...
"
```
